### ISSUE TITLE
Support or document suggestions for handling ConfigMaps

### ISSUE THREAD

**Issue by gtaylor** [helm/helm-classic#461]
I was trying to use helm to fill in some template variables from a central values.toml, but here's what I get when doing a `helm install` with a configmap in the manifests dir:
```
error validating "STDIN": error validating data: API version "" isn't supported, only supports API versions ["metrics/v1alpha1" "extensions/v1beta1" "componentconfig/v1alpha1" "batch/v1" "autoscaling/v1" "authorization.k8s.io/v1beta1" "v1"]; if you choose to ignore these errors, turn validation off with --validate=false
```
Here is the generated configmap:
```
#helm:generate helm tpl -d values.toml -o manifests/drone-configmap.yaml $HELM_GENERATE_FILE
apiVersion: v1
kind: ConfigMap
metadata:
  creationTimestamp: 2016-03-30T21:14:38Z
  name: drone
  namespace: default
data:
  caddy.conf: |-
    tls replace@me.com
    drone.testing.com {
        proxy / drone:80 {
          proxy_header X-Forwarded-Proto {scheme}
          proxy_header X-Forwarded-For {host}
          proxy_header Host {host}
        }
    }
```
My goal is to tell people to grab the chart, swap the variables out in values.toml, and install it. This is the smoothest user experience I've been able to come up with for my audience.

Is helm going to support generated configmaps, or are we going to require that they create/upload their own before running a helm chart?

**Comment by krancour** [comment]
That error is actually coming from kubectl (which helm uses).  kubectl does cache information about what APIs are supported by a cluster.  I think you've got info for an older cluster stuck in your cache.  I can't quite recall how to clear it.

**Comment by krancour** [comment]
Actually... I take that back.  Since configMap has moved to the v1 API, there should be no circumstance where that's not enabled on a cluster (and no circumstance where the cache of supported APIs says otherwise).  I'll dig into this.

**Comment by krancour** [comment]
Ok... configMap support is outright missing.  See the recent #462 which adds deployment support for an example of how support for configMap can be added.  If you'd like to PR it, it would be welcome.  (It should be installed/uninstalled around the same time as secrets.)

**Comment by krancour** [comment]
Per my explanation in #456 helm implicitly supports k8s 1.2, so I would regard lack of support for configMap as a bug-- which is somewhat important because this project is in maintenance mode since it will be eventually superseded by kubernetes/helm.  I'm updating the tags accordingly.

**Comment by mboersma** [comment]
@gtaylor the error you saw should be fixed by #479. You can verify with the current `helmc` client at  [ ![Download](https://api.bintray.com/packages/deis/helm-ci/helmc/images/download.svg) ](https://bintray.com/deis/helm-ci/helmc/_latestVersion#files).

My only question is does a `ConfigMap` have a natural [install ordering](https://github.com/helm/helm-classic/blob/master/action/install.go#L18) that should be enforced?

**Comment by krancour** [comment]
@mboersma, I believe as far as install order is concerned, config maps would be level with secrets.
